pkg/controller/shared_state.go

```diff
@@ -1176,9 +1178,14 @@ func (c *SharedState) createReplicas(ctx context.Context, remainingReplicas int,
 			//Push to queue the failed replica number
 			request := ReplicaRequest{VolumeName: volumeName, Priority: remainingReplicas}
 			c.priorityReplicaRequestsQueue.Push(ctx, &request)
```

Comment:
If we exit the loop and `remainingReplicas > 0`, a `ReplicaRequest` will be pushed to the operation queue on line 1194 below, so we do not need to do this here.

Reply:
Done.

Reply:
Done. Yes, we shouldn't push the failed replica to the queue here, since we will keep trying next node. I missed it here and make the code so confused. Now the logic is clean!